subsys/bluetooth/host/l2cap_internal.h

```diff
@@ -360,3 +360,12 @@ void l2cap_br_encrypt_change(struct bt_conn *conn, uint8_t hci_status);
 
 /* Handle received data */
 void bt_l2cap_br_recv(struct bt_conn *conn, struct net_buf *buf);
+
+struct bt_l2cap_cb {
```

Comment:
Is this some kind of ecred callback? Based on the name it sounds like some much more general callback. Please consider renaming it, eg `bt_l2cap_ecred_cb`

Btw, is it intentional that only one instance is allowed. Is it possible some other protocol besides ATT will want to register such callbacks too?

Reply:
Currently it is only used by ecred connection, so could rename it. If anyone adds more callbacks in the future the name can be changed back to the more general one.

I only added one instance because that is what I needed. Could also use a list like it is done in GATT.

Reply:
Kept one instance and renamed to only be ecred callbacks for now. Multiple callbacks or callbacks for non-ecred stuff can be added if needed in the future.